tests/unit/test_dataset_preprocessor.py

```diff
@@ -201,6 +201,220 @@ def test_preprocess(mock_get_dataset: MagicMock, dataset_rule_equal_to: dict):
     assert preprocessed_dataset.equals(expected_dataset)
 
 
+@patch("cdisc_rules_engine.services.data_services.LocalDataService.get_dataset")
+def test_preprocess_left_join(mock_get_dataset: MagicMock, dataset_rule_equal_to: dict):
```

Comment:
This looks the same as `test_preprocess`, except for the addition of `"join_type": "left"` and the expected output. I think it would be better to use the same function, but parameterize the input datasets, expected output, and join type. This way you aren't repeating the input datasets and the code.

Reply:
I've updated `test_preprocess` to accept `join_type` and `expected_dataset` as parameters (testing for `join_type` values of `None`, `"inner"` and `"left"`), and have removed `test_preprocess_left_join`.

I've also renamed `test_merge_datasets_on_left_join` to `test_merge_datasets_on_join_type` and parameterized it too to test for correct functioning for both `join_type = JoinTypes.INNER` and `join_type = JoinTypes.LEFT`